evennia/contrib/pathfinding/pathfinder.py

```diff
@@ -0,0 +1,212 @@
+        """
+        Updates the graph, adding or deleting any new/old Rooms and Exits.
+
+        Returns:
+            graph (DiGraph): Networkx directional graph of your game's
+                rooms and exits.
+
+        """
+        graph = self._graph
+        
+        # Get all Room objects in play
+        all_rooms = Room.objects.all().iterator()
+        
+        for src in all_rooms:
+            # Create node for src room
+            src_key = self.get_key(src)
+            
+            # Check if node exists
+            try: graph[src_key]
+            except: graph.add_node(src_key, key=src.db_key, id=src.id, type='room')
+            
+            for exit in src.exits:
+                # Get destination room
+                dst = exit.destination
+                
+                # Create a node for dst room
+                dst_key = self.get_key(dst)
+                try: graph[dst_key]
+                except: graph.add_node(dst_key, key=dst.db_key, id=dst.id, type='room')
+                
+                # Create an edge representing the exit
+                graph.add_edge(src_key, dst_key, direction=exit.db_key, id=exit.id, type='exit')
+                
+        return graph
+                
+    def get_path(self, source, dest):
+        """
+        Computes shortest path between source and dest objects.
+        
+        This is primarily an internal function; it is recommended that you use 
+        one of the other helper methods to perform any queries.
+        
+        Args:
+            source (Room): Origin Room object.
+            dest (Room): Destination Room object.
+
+        Returns:
+            path (list): List of node tokens comprising the path from the source
+                to the detination.
+
+        """
+        # Get keys for source and dest
+        src_key = self.get_key(source)
+        dst_key = self.get_key(dest)
+        
+        # Get the shortest path
+        path = nx.shortest_path(self._graph, source=src_key, target=dst_key)
+        
+        return path
+        
+    def get_directions(self, source, dest):
+        """
+        Computes shortest path between source and dest objects.
+        
+        Returns a list of the directional commands you would need to execute to 
+        move from your starting location to your desired destination.
+        
+        Args:
+            source (Room): Origin Room object.
+            dest (Room): Destination Room object.
+
+        Returns:
+            steps (list): List of directional commands.
+
+        """
+        path = self.get_path(source, dest)
+        steps = []
+        
+        # Get the edge attributes for each hop
+        edge_ids = zip(path[0:], path[1:])
+        for src, dst in edge_ids:
+            direction = self._graph[src][dst]['direction']
+            steps.append(direction)
+        
+        return steps
+        
+    def get_line_of_sight(self, source, direction, ordered=True):
```

Comment:
This appears to get all rooms in a line. Would not a `max_distance` argument be a useful and practical optimization for this method?

Reply:
Yeah, that'll break it. Nice catch :)

A distance arg would be useful too.